src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -73,10 +73,44 @@ public OnnxNodeInfo(string name, OnnxShape shape, System.Type type)
         public readonly List<string> InputNames;
         public readonly List<string> OutputNames;
 
+        /// <summary>
+        /// Constructs OnnxModel object from file.
+        /// </summary>
+        /// <param name="modelFile">Model file path</param>
         public OnnxModel(string modelFile)
+            : this (modelFile, -1, false)
+        {
+        }
+
+        /// <summary>
+        /// Constructs OnnxModel object from file.
+        /// </summary>
+        /// <param name="modelFile">Model file path</param>
+        /// <param name="gpuDeviceId">Gpu device ID to execute on. -1 for CPU</param>
+        /// <param name="fallbackToCpu">If true, resumes CPU execution quitely upon GPU error</param>
+        public OnnxModel(string modelFile, int gpuDeviceId = -1, bool fallbackToCpu = false)
```

Comment:
>  public OnnxModel(string modelFile, int gpuDeviceId = -1, bool fallbackToCpu = false) [](start = 6, length = 86)

same comment about ctor redundance

Reply:
+1, the default arguments should get the same result of OnnxModel(string modelFile).

---
In reply to: [243430419](https://github.com/dotnet/machinelearning/pull/1922#discussion_r243430419) [](ancestors = 243430419)

Reply:
Removed redundant ctor.